### ISSUE TITLE
[Feat]: A generic "sql-query" collector

### ISSUE THREAD

**Issue by andrewm4894** [netdata/netdata#13659]
### Problem

I have data in an SQL database that i'd like to chart and monitor in Netdata. I want to be able to configure a netdata collector to make a query against this data to update a chart in netdata.

This would open up more general/generic data monitoring capabilities in netdata.

### Description

I want to be able to define a query to run against some sql database where each query returns data for a chart in netdata. 

This is like a higher level abstraction of work i did for [this proposed bigquery collector](https://github.com/netdata/netdata/pull/13599). Basically dont just solve it for BigQuery but implement something more generic that could qry against BigQuery, SnowFlake, Redshift, ClickHouse, MySql basically any standard enough sql complaint database.

Obviously we could just start with one or two supported targets like BigQuery and MySql for example and add from there. 

For example, for our own use case we have product telemetry events flowing in real time into BigQuery. I would like to be able to define batches of sql statements to make charts for our key busienss metrics derived from those raw telemetry events on a rolling window basis.

### Importance

really want

### Value proposition

1. Ability to monitor anything i can write a sql query for.
2.
3.
...

### Proposed implementation

Some collector that can take in a generic config and then take advantage of sql standards that a lot of databases use for interoperability. Perhaps using standards like jdcs or some such cross platform extendabile approach to make things as easy as possible for each new target sql db.

**Comment by shyamvalsan** [comment]
@cakrit @amalkov @sashwathn might be a good addition to the use-case list. This one is about monitoring the data that the database contains, rather than the database performance. And it can be generic to any SQL based database.

-  "How to monitor your Snowflake data using Netdata"
- "How to monitor your BigQuery data using Netdata"
- etc. 

Not to mention we could be dogfooding our own internal KPIs using Netdata. Maybe eventually replace the Grafana dashboard we use to monitor connected nodes etc.

**Comment by cakrit** [comment]
Really nice idea!

**Comment by ilyam8** [comment]
I didn't really understand anything except this general idea

> Ability to monitor anything i can write a sql query for.

For me, it is user-defined SQL queries ([e.g.](https://github.com/CrunchyData/pgmonitor/blob/a65c7cf1aa62620935a7abcf3fe9f583dff41a20/postgres_exporter/common/pg14/queries_general.yml)) data from which (labels and values) is used to create user-defined charts. Where _anything_ is limited to existing go SQL drivers.

An example of such functionality is telegraf's [sql input plugin](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/sql).

**Comment by shyamvalsan** [comment]
@ilyam8 yes, this is similar to the telegraf plugin you linked to. Basically, Netdata can become a nice monitoring platform with built in anomaly detection capability for data that can be retrieved via SQL.

**Comment by amalkov** [comment]
We already discussed this suggestion with @andrewm4894 and I agree, it would be really helpful to have it.

PS: probably it would be good to set default collect every to something big, for example 10 mins or even 1 hour. 

For people to think first before changing it as it will add an extra load to the DB.

**Comment by shyamvalsan** [comment]
Agree - default interval can be something higher, user can set it lower if they want to.

**Comment by ilyam8** [comment]
If the default value is 1hr+ - a person will open a bug report earlier than the chart will appear on the dashboard (up to 2 hrs) 🤣 Anyway, that is a minor technical detail not really worth discussing in my opinion.

**Comment by andrewm4894** [comment]
Yeah interval maybe should have to be part of the config of each query or job.

**Comment by ilyam8** [comment]
> a minor technical detail not really worth discussing in my opinion.
> part of the config of each query or job.

Yes, I had this in mind (because goes without saying) and was wondering why to bother.

**Comment by shyamvalsan** [comment]
This exists now with Pandas.

**Comment by andrewm4894** [comment]
I think idea is it would maybe be more suitable to graduate this from being based on pandas python collector to a more first class citizen and fully supported Go based plugin. 

The pandas approach is ok but not ideal as involves some dependencies and things like that. 

I'd suggest we reopen. What do you think @ilyam8 ?
